### ISSUE TITLE
Reports being run repeatedly

### ISSUE THREAD

**Issue by azvoleff** [ConservationInternational/trends.earth#648]
@gkahiu - I've noticed that the reports seem to get repeatedly run - messages pop up in the QGIS message bar saying that reports are running a few times a minute, even when the reports are already finished for a particular job. There are also a good number of warnings in the python warnings tab of the message bar (they are all identical so I've just posted two here):

```
2022-05-12T11:24:55     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:25:03     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:25:19     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:25:27     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:25:44     INFO    Downloading D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\data\admin_bounds_polys_COL.json.gz
2022-05-12T11:26:00     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:26:07     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:26:16     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:26:25     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:28:10     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:28:21     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:29:24     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:29:27     INFO    Authentication Manager : Master password has been successfully read from your Password Manager
2022-05-12T11:30:07     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:30:22     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:30:42     INFO    Submitted : Drought indicator task submitted to Google Earth Engine.
2022-05-12T11:31:03     INFO    Downloading D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\data\admin_bounds_polys_CIV.json.gz
2022-05-12T11:31:09     INFO    Submitted : Drought indicator task submitted to Google Earth Engine.
2022-05-12T11:31:11     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:31:51     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:32:31     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:13     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:22     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:28     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:38     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:47     INFO    Report Status : botswana-francistown reports are being processed...
2022-05-12T11:33:57     INFO    Report Status : botswana-francistown reports are being processed...
```

There are also a good number of warnings in the python warnings tab of the log messages window that appear to be releated (they are all identical so I've just posted two here):

```
2022-05-12T11:35:59     WARNING    warning:C:\OSGeo4W\apps\Python39\lib\subprocess.py:1052: ResourceWarning: subprocess 40696 is still running
              _warn("subprocess %s is still running" % self.pid,
             
             traceback: File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 1227, in on_task_status_changed
              self.remove_task_context(ctx)
              File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 1176, in remove_task_context
              handler.cancel()
              File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 965, in cancel
              subprocess.Popen(f'TASKKILL /F /PID {pid} /T', shell=True)
              File "C:\OSGeo4W\apps\Python39\lib\subprocess.py", line 1052, in __del__
              _warn("subprocess %s is still running" % self.pid,
             
2022-05-12T11:36:41     WARNING    warning:C:\OSGeo4W\apps\Python39\lib\subprocess.py:1052: ResourceWarning: subprocess 15644 is still running
              _warn("subprocess %s is still running" % self.pid,
             
             traceback: File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 1227, in on_task_status_changed
              self.remove_task_context(ctx)
              File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 1176, in remove_task_context
              handler.cancel()
              File "D:\Documents and Settings/azvoleff/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\reports\generator.py", line 965, in cancel
              subprocess.Popen(f'TASKKILL /F /PID {pid} /T', shell=True)
              File "C:\OSGeo4W\apps\Python39\lib\subprocess.py", line 1052, in __del__
              _warn("subprocess %s is still running" % self.pid,
             
```

**Comment by gkahiu** [comment]
@azvoleff, for which algorithm and country/region are you getting this error?

**Comment by azvoleff** [comment]
It is for SDG 15-3-1 for Botswana, subregion Francistown. The zip of the data folder [is here](https://github.com/ConservationInternational/trends.earth/files/8686748/d7a36299-c94a-466c-90ce-b61ad3a9e7e7.zip).
.

**Comment by gkahiu** [comment]
Can't reproduce, maybe its related to the thread crash issue...?

Could you enable report logs in the settings dialog:

![image](https://user-images.githubusercontent.com/3840267/168582957-06062279-9c62-4540-a047-5602894504c6.png)

Then copy the contents of the contents of the html file in the reports folder?

![image](https://user-images.githubusercontent.com/3840267/168583431-2569bdec-83ca-4ad5-81a3-4168d5c01187.png)

**Comment by azvoleff** [comment]
Thanks - I've tracked this down. It is related to [a change in trends.earth-algorithms](https://github.com/ConservationInternational/trends.earth-algorithms/commit/9ba0a15dcb3bc42c7d71b6e7b88bf455c95b0a08) that led to a parameter getting dropped that is needed for the reports. This one is on me!

Thanks for pointing out the report logging option, I hadn't seen that - the html let me figure it out.

**Comment by azvoleff** [comment]
Fixed by 1bb6f114
